File: youtask-fe/src/services/AuthService.ts
import http from "../http-common";
import type { AuthRequest, AuthResponse, UserRequest } from "../types/index";

const login = (data: AuthRequest) => {
  return http.post<AuthResponse>("/auth/login", data);
};

const register = (data: UserRequest) => {
  return http.post<AuthResponse>("/auth/register", data);
};

const logout = () => {
  localStorage.removeItem("accessToken");
  return Promise.resolve();
};

const AuthService = {
  login,
  register,
  logout,
};

export default AuthService;

File: youtask-general-api/src/main/java/com/med/youtaskgeneralapi/service/AuthService.java
package com.med.youtaskgeneralapi.service;

import com.med.youtaskgeneralapi.model.dto.AuthRequest;
import com.med.youtaskgeneralapi.model.dto.AuthResponse;
import com.med.youtaskgeneralapi.model.dto.UserRequest;

public interface AuthService {

    AuthResponse register(UserRequest UserRequest);
    AuthResponse login(AuthRequest authRequest);

}
